src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/Body.js

```diff
@@ -25,163 +21,39 @@ type Props = {
-        } else if (this.isMultipleSelect()) {
-            prependedCells.push(this.createCheckboxCell(rowProps, rowIndex));
-        }
-
-        const clonedCells = this.cloneCells(cells, rowIndex);
-
-        clonedCells.unshift(...prependedCells);
-
-        return clonedCells;
-    };
-
-    cloneCells = (originalCells: any, rowIndex: number) => {
-        return React.Children.map(originalCells, (cell, index) => {
-            if (React.isValidElement(cell)) {
-                return React.cloneElement(
-                    cell,
-                    {
-                        key: `body-cell-${rowIndex}-${index}`,
-                    }
-                );
-            }
-        });
-    };
-
-    createRadioCell = (rowProps: RowProps, rowIndex: number) => {
-        const key = `body-radio-${rowIndex}`;
-        const identifier = rowProps.id || rowIndex;
-
-        return (
-            <Cell
-                key={key}
-                isSmall={true}>
-                <Radio
-                    skin="dark"
-                    value={identifier}
-                    checked={!!rowProps.selected}
-                    onChange={this.handleRowSingleSelectionChange} />
-            </Cell>
-        );
-    };
-
-    createCheckboxCell = (rowProps: RowProps, rowIndex: number) => {
-        const key = `body-checkbox-${rowIndex}`;
-        const identifier = rowProps.id || rowIndex;
-
-        return (
-            <Cell
-                key={key}
-                isSmall={true}>
-                <Checkbox
-                    skin="dark"
-                    value={identifier}
-                    checked={!!rowProps.selected}
-                    onChange={this.handleRowMultipleSelectionChange} />
-            </Cell>
-        );
-    };
-
-    createControlCells = (rowProps: RowProps, rowIndex: number) => {
-        const {controls} = this.props;
-
-        if (!controls) {
-            return null;
-        }
-
-        return controls.map((controlItem: ControlConfig, index) => {
-            const key = `body-control-${rowIndex}-${index}`;
-            const handleControlClick = () => {
-                controlItem.onClick(rowProps.id || rowIndex);
-            };
-
-            return (
-                <Cell
-                    key={key}
-                    isControl={true}>
-                    <button onClick={handleControlClick}>
-                        <Icon name={controlItem.icon} />
-                    </button>
-                </Cell>
+            return React.cloneElement(
+                row,
+                {
+                    ...row.props,
+                    key: `body-row-${index}`,
+                    rowIndex: index,
+                    buttons: buttons,
+                    selectMode: selectMode,
+                    onSingleSelectionChange: this.handleRowSingleSelectionChange,
+                    onMultipleSelectionChange: this.handleRowMultipleSelectionChange,
```

Comment:
Can't the previous two be combined into one somehow?

Reply:
I could probably do that by also setting the first function argument of the single select to a boolean (checked). But idk if that would be a good solution